libpod/oci_conmon_common.go

```diff
@@ -371,7 +371,17 @@ func (r *ConmonOCIRuntime) KillContainer(ctr *Container, signal uint, all bool)
 	}
 	if err := utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, env, r.path, args...); err != nil {
 		// Update container state - there's a chance we failed because
-		// the container exited in the meantime.
+		// the container exited in the meantime. Note that we need to
+		// take the lock and resync the container.
+		if !ctr.batched {
+			ctr.lock.Lock()
```

Comment:
This lock is already locked in https://github.com/containers/podman/blob/bc7695a3b297e89b6933f58369d17be8377c199b/libpod/container_api.go#L220 and this is a deadlock. Am I wrong?

Reply:
Sorry, dup